### PR TITLE
fix(teams): resolve undefined variable error and add logging

### PR DESCRIPTION
**Pull request by miaulalala**
Signed-off-by: Anna Larch <anna@nextcloud.com>

### CODE REVIEW

File: lib/Controller/ContactController.php [hunk 1]
```diff
@@ -173,32 +178,32 @@ public function searchAttendee(string $search):JSONResponse {
...
 	 * @return JSONResponse
 	 * @throws Exception
-	 * @throws \OCP\AppFramework\QueryException
 	 *
 	 * @NoAdminRequired
 	 */
 	public function getCircleMembers(string $circleId):JSONResponse {
-		if (!$this->appManager->isEnabledForUser('circles') || !class_exists('\OCA\Circles\Api\v1\Circles')) {
+		if (!class_exists('\OCA\Circles\Api\v1\Circles') || !$this->appManager->isEnabledForUser('circles')) {
+			$this->logger->debug('Circles not enabled');
```
Comment (kesselb) [inline lib/Controller/ContactController.php:186]: Why the debug log?

Comment (miaulalala) [reply lib/Controller/ContactController.php:186]: Why not?

File: lib/Controller/ContactController.php [hunk 2]
```diff
@@ -173,32 +178,32 @@ public function searchAttendee(string $search):JSONResponse {
 	 * @param string $circleId CircleId to query for members
 	 * @return JSONResponse
 	 * @throws Exception
-	 * @throws \OCP\AppFramework\QueryException
 	 *
 	 * @NoAdminRequired
 	 */
 	public function getCircleMembers(string $circleId):JSONResponse {
-		if (!$this->appManager->isEnabledForUser('circles') || !class_exists('\OCA\Circles\Api\v1\Circles')) {
+		if (!class_exists('\OCA\Circles\Api\v1\Circles') || !$this->appManager->isEnabledForUser('circles')) {
```
Comment (SebastianKrupinski) [inline lib/Controller/ContactController.php:185]: Question, why are we checking if the class exists? Would the check for isEnabledForUser not be sufficient enough to  determine if the circles app is installed, enabled on the system and permitted for this user?

Comment (kesselb) [reply lib/Controller/ContactController.php:185]: The original pr mentions a problem with Psalm https://github.com/nextcloud/calendar/pull/5420#pullrequestreview-1710099584 ;)

I haven't done any profiling but would expect, especially now with the flipped condition, a little performance gain for systems without circles. 

isEnabledForUser does a lot (fetch the installed app versions for all apps, see if the app is installed, fetch the user session, fetch the groups for a user, see if the user has a group that the app is enabled for).

Comment (SebastianKrupinski) [reply lib/Controller/ContactController.php:185]: > isEnabledForUser does a lot (fetch the installed app versions for all apps, see if the app is installed, fetch the user session, fetch the groups for a user, see if the user has a group that the app is enabled for).

I checked the code... Thats not the case, I think the original author just didn't know there was a build in function to see if the app was enabled, and checked for the existence for the class.

**Review by SebastianKrupinski — APPROVED** [review]

File: lib/Controller/ContactController.php (hunk 2 above)
Comment (kesselb) [reply lib/Controller/ContactController.php:185]: > Thats not the case

What is not the case?

File: lib/Controller/ContactController.php (hunk 1 above)
Comment (kesselb) [reply lib/Controller/ContactController.php:186]: The method is only called, from the frontend, when circles is enabled.

Comment (miaulalala) [reply lib/Controller/ContactController.php:187]: apparently not otherwise we wouldn't need the class_exists check / app enabled check

**Pull request merged**